server/bleep/src/webserver/repos.rs

```diff
@@ -230,10 +230,12 @@ pub(super) async fn index_status(Extension(app): Extension<Application>) -> impl
     let mut receiver = app.sync_queue.subscribe();
 
     Sse::new(async_stream::stream! {
-        while let Ok(event) = receiver.recv().await {
-            yield sse::Event::default().json_data(event).map_err(|err| {
-                <_ as Into<Box<dyn std::error::Error + Send + Sync>>>::into(err)
-            });
+        loop {
+            if let Ok(event) = receiver.recv().await {
+                yield sse::Event::default().json_data(event).map_err(|err| {
+                    <_ as Into<Box<dyn std::error::Error + Send + Sync>>>::into(err)
+                });
```

Comment:
Can we simplify this?

```suggestion
                yield sse::Event::default().json_data(event).map_err(Box::new);
```